go/pkg/tool/embeddedtool.go

```diff
@@ -0,0 +1,162 @@
+// the operation.
+var RemoteToolOperations = map[OpType]func(ctx context.Context, c *Client){
+	downloadActionResult: func(ctx context.Context, c *Client) {
+		if err := c.DownloadActionResult(ctx, getDigestFlag(), getPathFlag()); err != nil {
+			log.Exitf("error downloading action result for digest %v: %v", getDigestFlag(), err)
+		}
+	},
+	downloadBlob: func(ctx context.Context, c *Client) {
+		res, err := c.DownloadBlob(ctx, getDigestFlag(), getPathFlag())
+		if err != nil {
+			log.Exitf("error downloading blob for digest %v: %v", getDigestFlag(), err)
+		}
+		os.Stdout.Write([]byte(res))
+	},
+	downloadDir: func(ctx context.Context, c *Client) {
+		if err := c.DownloadDirectory(ctx, getDigestFlag(), getPathFlag()); err != nil {
+			log.Exitf("error downloading directory for digest %v: %v", getDigestFlag(), err)
+		}
+	},
+	showAction: func(ctx context.Context, c *Client) {
+		res, err := c.ShowAction(ctx, getDigestFlag())
+		if err != nil {
+			log.Exitf("error fetching action %v: %v", getDigestFlag(), err)
+		}
+		os.Stdout.Write([]byte(res))
+	},
+	downloadAction: func(ctx context.Context, c *Client) {
+		err := c.DownloadAction(ctx, getDigestFlag(), getPathFlag(), overwrite)
+		if err != nil {
+			log.Exitf("error fetching action %v: %v", getDigestFlag(), err)
+		}
+		fmt.Printf("Action downloaded to %v\n", getPathFlag())
+	},
+	executeAction: func(ctx context.Context, c *Client) {
+		if _, err := c.ExecuteAction(ctx, getDigestFlag(), actionRoot, getPathFlag(), outerr.SystemOutErr); err != nil {
+			log.Exitf("error executing action: %v", err)
+		}
+	},
+	checkDeterminism: func(ctx context.Context, c *Client) {
+		if err := c.CheckDeterminism(ctx, getDigestFlag(), actionRoot, execAttempts); err != nil {
+			log.Exitf("error checking determinism: %v", err)
+		}
+	},
+	uploadBlob: func(ctx context.Context, c *Client) {
+		if err := c.UploadBlob(ctx, getPathFlag()); err != nil {
+			log.Exitf("error uploading blob for digest %v: %v", getDigestFlag(), err)
+		}
+	},
+	uploadBlobV2: func(ctx context.Context, c *Client) {
+		if err := c.UploadBlobV2(ctx, getPathFlag()); err != nil {
+			log.Exitf("error uploading blob for digest %v: %v", getDigestFlag(), err)
+		}
+	},
+	uploadDir: func(ctx context.Context, c *Client) {
+		us, err := c.UploadDirectory(ctx, getPathFlag())
+		if jsonOutput != "" {
+			js, _ := json.MarshalIndent(us, "", "  ")
+			if jsonOutput == "-" {
+				fmt.Printf("%s\n", js)
+			} else {
+				log.Infof("Outputting JSON results to %s", jsonOutput)
+				if err := os.WriteFile(jsonOutput, []byte(js), 0o666); err != nil {
+					log.Exitf("Error writing JSON output to file: %v", err)
+				}
+			}
+		}
+		if err != nil {
+			log.Exitf("error uploading directory for path %s: %v", getPathFlag(), err)
+		}
+	},
+}
+
+// ValidateFlags validates the command line flags associated with this embedded remote tool.
+func ValidateFlags() {
+	if !flag.Parsed() {
+		flag.Parse()
```

Comment:
I think should not be done here now that the flags are the responsibility of the user.

Reply:
Done